opnode/l2/driver.go

```diff
@@ -2,88 +2,203 @@ package l2
 
 import (
 	"context"
+	"encoding/binary"
 	"fmt"
+	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/trie"
 	"github.com/holiman/uint256"
 )
 
-type InputItem struct {
-	Block    *types.Block
-	Receipts []*types.Receipt
-}
+var (
+	DepositEventABI     = "TransactionDeposited(address,address,uint256,uint256,bool,bytes)"
+	DepositEventABIHash = crypto.Keccak256Hash([]byte(DepositEventABI))
+	DepositContractAddr = common.HexToAddress("0xdeaddeaddeaddeaddeaddeaddeaddeaddead0001")
+	L1InfoFuncSignature = "setL1BlockValues(uint256 _number, uint256 _timestamp, uint256 _basefee, bytes32 _hash)"
+	L1InfoFuncBytes4    = crypto.Keccak256([]byte(L1InfoFuncSignature))[:4]
+	L1InfoPredeployAddr = common.HexToAddress("0x4242424242424242424242424242424242424242")
+)
 
-func (item *InputItem) Epoch() uint64 {
-	return item.Block.NumberU64()
-}
+// UnmarshalLogEvent decodes an EVM log entry emitted by the deposit contract into typed deposit data.
+// parse log data for:
+//     event TransactionDeposited(
+//    	 address indexed from,
+//    	 address indexed to,
+//    	 uint256 value,
+//    	 uint256 gasLimit,
+//    	 bool isCreation,
+//    	 data data
+//     );
+func UnmarshalLogEvent(blockNum uint64, txIndex uint64, ev *types.Log) (*types.DepositTx, error) {
+	if len(ev.Topics) != 3 {
+		return nil, fmt.Errorf("expected 3 event topics (event identity, indexed from, indexed to)")
+	}
+	if ev.Topics[0] != DepositEventABIHash {
+		return nil, fmt.Errorf("invalid deposit event selector: %s, expected %s", ev.Topics[0], DepositEventABIHash)
+	}
+	if len(ev.Data) < 160 {
+		return nil, fmt.Errorf("deposit event data too small (%d bytes): %x", len(ev.Data), ev.Data)
+	}
```

Comment:
Are all those sanity checks really necessary? What are your trust assumptions regarding the input (might be worth documenting that)?